### PR TITLE
Create a log wrapper and better test GDLLogWriter

### PR DESCRIPTION
**Pull request by mikehaney24**
Hey there! So you want to contribute to a Firebase SDK?
Before you file this pull request, please read these guidelines:

### Discussion

  * Read the contribution guidelines (CONTRIBUTING.md).
  * If this has been discussed in an issue, make sure to link to the issue here.
    If not, go file an issue about this **before creating a pull request** to discuss.

### Testing

  * Make sure all existing tests in the repository pass after your change.
  * If you fixed a bug or added a feature, add a new test to cover your code.

### API Changes

  * At this time we cannot accept changes that affect the public API.  If you'd like to help
    us make Firebase APIs better, please propose your change in a feature request so that we
    can discuss it together.

### CODE REVIEW

File: GoogleDataLogger/GoogleDataLogger/DependencyWrappers/GDLConsoleLogger.h [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+/** The console logger prefix. */
+static GULLoggerService kGDLConsoleLogger = @"[GoogleDataLogger]";
+
+/** A list of message codes to print in the logger that help to correspond printed messages with
+ * code locations. */
+typedef NS_ENUM(NSInteger, GDLMessageCode) {
+
+  /** For warning messages concerning transform: not being implemented by a log transformer. */
+  GDLMCWTransformerDoesntImplementTransform = 1
```
Comment (paulb777) [inline GoogleDataLogger/GoogleDataLogger/DependencyWrappers/GDLConsoleLogger.h]: What does MCW stand for?

File: GoogleDataLogger/GoogleDataLogger/Classes/GDLLogEvent.h [hunk 2]
```diff
@@ -44,6 +44,9 @@ typedef NS_ENUM(NSInteger, GDLLogQoS) {
 /** The log map identifier, to allow backends to map the extension property to a proto. */
 @property(readonly, nonatomic) NSString *logMapID;
 
+/** The identifier for the backend this log will eventually be sent to. */
+@property(readonly, nonatomic) NSInteger logTarget;
```
Comment (paulb777) [inline GoogleDataLogger/GoogleDataLogger/Classes/GDLLogEvent.h]: Why is an integer needed in addition to the logMapID?

File: GoogleDataLogger/GoogleDataLogger/DependencyWrappers/GDLConsoleLogger.h (hunk 1 above)
Comment (mikehaney24) [reply GoogleDataLogger/GoogleDataLogger/DependencyWrappers/GDLConsoleLogger.h]: MessageCodeWarning. I'm trying to keep these strings short. Any alternative suggestions?

File: GoogleDataLogger/GoogleDataLogger/Classes/GDLLogEvent.h (hunk 2 above)
Comment (mikehaney24) [reply GoogleDataLogger/GoogleDataLogger/Classes/GDLLogEvent.h]: the logMapID is for the backend to use to map a given log extension to the appropriate proto.
the logTarget is an identifier, probably defined by a future shared proto, that defines the backend this log should be sent to.

File: GoogleDataLogger/GoogleDataLogger/DependencyWrappers/GDLConsoleLogger.h (hunk 1 above)
Comment (paulb777) [reply GoogleDataLogger/GoogleDataLogger/DependencyWrappers/GDLConsoleLogger.h]: Probably fine with a comment.

**Pull request merged**